src/backend/joanie/core/models/contracts.py

```diff
@@ -0,0 +1,163 @@
+"""
+Declare and configure the models for Joanie's contracts
+"""
+import hashlib
+import logging
+
+from django.conf import settings
+from django.core.exceptions import PermissionDenied
+from django.db import models
+from django.utils.functional import lazy
+from django.utils.module_loading import import_string
+from django.utils.translation import gettext_lazy as _
+
+from joanie.core.utils import image_to_base64
+
+from .accounts import User
+from .base import BaseModel
+from .products import Order
+
+logger = logging.getLogger(__name__)
+
+
+class ContractDefinition(BaseModel):
+    """
+    Contract definition describes template and markdown to generate user contracts
+    """
+
+    title = models.CharField(_("title"), max_length=255)
+    body = models.TextField(_("body"), blank=True)
+    language = models.CharField(
+        max_length=10,
+        choices=lazy(lambda: settings.LANGUAGES, tuple)(),
+        verbose_name=_("language"),
+        help_text=_("Language of the contract definition"),
+    )
```

Comment:
Yes. Most contract definitions will be created only in one language. If a contract definition is created in several languages, it may or may not be in a language of the site?